### PR TITLE
Remove ANTSPATH

### LINKED ISSUE #1573: Do we still need ANTSPATH?
With ANTs now available for Conda, I was wondering whether there is any use for `ANTSPATH`?

Right now, some bash scripts cannot be run without explicitly setting `ANTSPATH` to the Conda env bin directory, despite the guarantee that all binaries will be present under `PATH`. I am actually wondering what use cases for `ANTSPATH` are not already covered by `PATH` overrides? 

I am tempted to patch all scripts to strip `ANTSPATH` from the Conda package, but it looks like other distributions of ANTs, like Homebrew, Debian or RPM packages, could benefit from this change too.

WDYT?

### PR DESCRIPTION
**Pull request by ghisvail**
Closes #1573

### CODE REVIEW

**Comment by ghisvail** [comment]
Looks like I am missing a few files (32 vs 34 in the search).

**Comment by cookpa** [comment]
Most of the scripts still need some work - removing ANTSPATH references is not sufficient because many of the checks for programs will fail. 

Eg

`-s "${ANTSPATH}ANTS"`

can't be replaced with

`-s "ANTS"`

but rather needs to be `-s $(which ANTS)`.

Is it OK if I push to your branch?

**Comment by ghisvail** [comment]
Sure, go for it 👍

Le mar. 25 juil. 2023, 13:43, Philip Cook ***@***.***> a
écrit :

**Comment by ghisvail** [comment]
> POSIX check for a command is `command -v` as which may not always be available

PR updated with `if ! command -v $CMD &> /dev/null` guards as per this SO [thread](https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script).

File: Scripts/ants.sh [hunk 1]
```diff
@@ -3,20 +3,17 @@
...
 
 MAXITERATIONS=30x90x20
-if [[ -z ${ANTSPATH} ]] ; then
-  echo "Environment variable ANTSPATH must be defined"
-  exit 1
-fi
-if [[ ! -f "${ANTSPATH}/ANTS" ]] ; then
-  echo "Cannot find the ANTS program. Please \(re\)define \$ANTSPATH in your environment."
+
+if ! command -v "ANTS" &> /dev/null
```
Comment (cookpa) [inline Scripts/ants.sh:7]: I think you need the square brackets still

Comment (ghisvail) [reply Scripts/ants.sh:7]: It works without brackets on my machine. I used the pattern in the accepted answer.

Comment (ghisvail) [reply Scripts/ants.sh:7]:
```
# Make ANTS disappear from PATH.
$ mv $HOME/.local/ants/bin/ANTS{,.bak}

# Run wrapper script.
$ ./Scripts/ants.sh
Cannot find the ANTS program. Please \(re\)define $PATH in your environment.

# Put it back.
$ mv $HOME/.local/ants/bin/ANTS{.bak,}

# Run wrapper script again.
$ ./Scripts/ants.sh
 USAGE ::  
  sh   ants.sh  ImageDimension  fixed.ext  moving.ext  OPTIONAL-OUTPREFIX   OPTIONAL-max-iterations  OPTIONAL-Labels-In-Fixed-Image-Space-To-Deform-To-Moving-
```

**Comment by cookpa** [comment]
Looking good now @ghisvail

**Comment by cookpa** [comment]
Thanks for doing this @ghisvail

**Pull request merged**